src/pac_type.cc

```diff
@@ -411,14 +411,12 @@ void Type::GenBufferConfiguration(Output *out_cc, Env *env)
 		case BUFFER_BY_LENGTH:
 			if ( ! NeedsBufferingStateVar() )
  				break;
-		
-			if ( buffering_state_var_field_ )
-				{
```

Comment:
I took a quick look at the binpac code here, and I agree it seems it's indeed fine to skip these checks -- so it's a bit hard to tell for sure. Given the `NeedsBufferingStateVar()`, I'd expect `buffering_state_var_field_` to always be set here, so maybe we could do another `assert` on that? Or are the new asserts catching the same thing already?